scheduler/gform_backend.py

```diff
@@ -0,0 +1,395 @@
+"""
+Backend for reading and writing data with Google Sheets
+
+Reads form responses, dues payers.
+
+Writes cars with drivers and riders
+
+"""
+
+
+import logging
+import gspread
+import pandas as pd
+import json
+import util
+import datetime
+from random import uniform
+
+from classes.AuthorizedClient import AuthorizedClient
+from classes.WSCell import WSCell
+from classes.WSRange import WSRange
+
+
+logger = logging.getLogger(__name__)
+
+"""
+Specify column ids for these values. If the form changes, unlink the spreadsheet
+and create a new one. Then reset these values
+"""
+
+EMAIL_COLUMN = 1
+NAME_COLUMN = 2
+PHONE_COLUMN = 3
+CAR_DESCRIPTION_COLUMN = 6
+SEATS_COLUMN = 7
+IS_RIDER_COLUMN = 4
+IS_DRIVER_COLUMN = 5
+DAYS_INFO_START_COLUMN = 8
+
+
+"""
+Number of rows between cars in the output spreadsheet
+"""
+
+CAR_ROW_SPACING = 2
+
+
+def get_dues_payers(dues_sheet):
+    """
+    Gets the spreadsheet of dues payers
+    """
+
+    payers = set()
+    for entry in dues_sheet.get_all_records():
+        payers.add(entry["Uniqname"].strip())
+
+    return payers
+
+
+def validate_dues_payers(email, dues_payers):
+    """
+    Checks if email is a dues payer
+    """
+
+    uniqname = email.split("@")[0].strip()
+    return uniqname in dues_payers
+
+
+def parse_location(location):
+    """
+    Converts location name in value for locations list
+    """
+
+    if location == "North Campus (Pierpont Commons)":
+        return "NORTH"
+    elif location == "Central Campus (The Cube)":
+        return "CENTRAL"
```

Comment:
Add response types to config file. avoid having to change to code if we want to change the form a bit